string drivename
uint32 errorcode
